client/js/components/image-viewer.js

```diff
@@ -33,8 +33,16 @@ function setupViewer(imageInfoSrc, viewer, viewerId) {
   }).catch(err => { throw err; });
 }
 
+function showViewerOnPage(viewerContent) {
```

Comment:
I'd have been tempted to have an e.g. `setViewerOnPageShown(bool)` function that covers both shown and hidden cases.